src/binding/petsc4py/src/petsc4py/PETSc/MatPartitioning.pyx

```diff
@@ -23,38 +24,134 @@ cdef class MatPartitioning(Object):
     def __call__(self):
         return self.getValue()
 
-    def view(self, Viewer viewer=None):
+    def view(self, Viewer viewer=None) -> None:
+        """View the partitioning data structure.
+
+        Collective.
+
+        Parameters
+        ----------
+        viewer
+            A `Viewer` to display the graph.
+
+        See also
+        --------
+        petsc.MatPartitioningView
+
+        """
         assert self.obj != NULL
         cdef PetscViewer vwr = NULL
         if viewer is not None: vwr = viewer.vwr
         CHKERR( MatPartitioningView(self.part, vwr) )
 
-    def destroy(self):
+    def destroy(self) -> Self:
+        """Destroy the partitioning context.
+
+        Collective.
+
+        See also
+        --------
+        petsc.MatPartitioningDestroy, create
+
+        """
         CHKERR( MatPartitioningDestroy(&self.part) )
         return self
 
-    def create(self, comm=None):
+    def create(self, comm: Comm | None = None) -> Self:
+        """Create a partitioning context.
+
+        Collective.
+
+        Parameters
+        ----------
+        comm
+            The MPI communicator.
+
+        See also
+        --------
+        petsc.MatPartitioningCreate, destroy
+
+        """
         cdef MPI_Comm ccomm = def_Comm(comm, PETSC_COMM_DEFAULT)
         CHKERR( MatPartitioningCreate(ccomm, &self.part) )
         return self
 
-    def setType(self, matpartitioning_type):
+    def setType(self, matpartitioning_type: Type | str) -> None:
+        """Set the type of the partitioner to use.
+
+        Collective.
+
+        Parameters
+        ----------
+        matpartitioning_type
+            The partitioner type.
+
+        See also
+        --------
+        petsc.MatPartitioningSetType, getType
+
+        """
         cdef PetscMatPartitioningType cval = NULL
         matpartitioning_type = str2bytes(matpartitioning_type, &cval)
         CHKERR( MatPartitioningSetType(self.part, cval) )
 
-    def getType(self):
+    def getType(self) -> Type:
+        """Return the partitioning method.
+
+        Not collective.
+
+        See also
+        --------
+        petsc.MatPartitioningGetType, setType
+
+        """
         cdef PetscMatPartitioningType cval = NULL
         CHKERR( MatPartitioningGetType(self.part, &cval) )
         return bytes2str(cval)
 
     def setFromOptions(self):
+        """Set parameters in the partitioner from the options database.
+
+        Collective.
+
+        See also
+        --------
+        petsc.MatPartitioningSetFromOptions, petsc_options
```

Comment:
Here and probably in many other places. We decided that python links goes before C links.